zeroconf/_handlers.py

```diff
@@ -105,30 +106,45 @@ def _answer_non_ptr_query(self, msg: DNSIncoming, out: DNSOutgoing, question: DN
             for dns_address in service.dns_addresses():
                 out.add_additional_answer(dns_address)
 
-    def response(self, msg: DNSIncoming, unicast: bool) -> Optional[DNSOutgoing]:
+    def response(
+        self, msg: DNSIncoming, addr: Optional[str], port: int
+    ) -> Tuple[Optional[DNSOutgoing], Optional[DNSOutgoing]]:
         """Deal with incoming query packets. Provides a response if possible."""
-        if unicast:
-            out = DNSOutgoing(_FLAGS_QR_RESPONSE | _FLAGS_AA, multicast=False)
+        unicast_out = None
+        if port != _MDNS_PORT:
+            unicast_out = DNSOutgoing(_FLAGS_QR_RESPONSE | _FLAGS_AA, multicast=False)
             for question in msg.questions:
-                out.add_question(question)
-        else:
-            out = DNSOutgoing(_FLAGS_QR_RESPONSE | _FLAGS_AA)
+                unicast_out.add_question(question)
+
+        multicast_out = DNSOutgoing(_FLAGS_QR_RESPONSE | _FLAGS_AA)
 
         for question in msg.questions:
             if question.type == _TYPE_PTR:
                 if question.name.lower() == _SERVICE_TYPE_ENUMERATION_NAME:
-                    self._answer_service_type_enumeration_query(msg, out)
+                    if unicast_out:
+                        self._answer_service_type_enumeration_query(msg, unicast_out)
+                    self._answer_service_type_enumeration_query(msg, multicast_out)
                 else:
-                    self._answer_ptr_query(msg, out, question)
+                    if unicast_out:
+                        self._answer_ptr_query(msg, unicast_out, question)
```

Comment:
Its not very efficient to construct it twice but no worse then the current design.

Its going to need to be refactored again anyways to fix

https://github.com/jstasiak/python-zeroconf/issues/595
https://github.com/jstasiak/python-zeroconf/issues/464
https://github.com/jstasiak/python-zeroconf/issues/500